### PR TITLE
Area save read

### PR DESCRIPTION
**Pull request by riku04**
**Description**
SQFliteを用いた領域データの保存と読み出し機能を追加
MapBlocのスコープを広げて読み出しのviewから操作出来るようにした
Areaクラスのテストコード を追加した

  Fixes #

  **Type of change**

  - [ ] Bug fix (non-breaking change which fixes an issue)
  - [x] New feature (non-breaking change which adds functionality)
  - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - [ ] This change requires a documentation update

  **Checklist**

  - [ ] My code follows the style guidelines of this project
  - [ ] I have performed a self-review of my own code
  - [ ] I have commented my code, particularly in hard-to-understand areas
  - [ ] I have made corresponding changes to the documentation
  - [ ] My changes generate no new warnings
  - [ ] I have added tests that prove my fix is effective or that my feature works
  - [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by riku04** [comment]
@isobe-h

File: lib/src/blocs/map_bloc.dart [hunk 1]
```diff
@@ -166,7 +204,8 @@ class MapBloc extends Bloc{
         }
     });
 
-    readSavedArea();
+    //readSavedArea(Constants.DEFAULT_AREA_TABLE);
```
Comment (isobe-h) [inline lib/src/blocs/map_bloc.dart]: :-1:

File: lib/src/blocs/map_bloc.dart [hunk 2]
```diff
@@ -110,27 +107,64 @@ class MapBloc extends Bloc{
...
+  void readSavedArea(String tableName) async{
+    AreaRepository().getPointsListFromTableName(tableName).then((areaList){
+      if(areaList.isEmpty){
+        return;
+      }
+      _polygons.clear();
+      areaList.forEach((points){
+        Polygon polygon = new Polygon(
+            points: points,
+            color: Color(Colors.red.value-0x20000000),
```
Comment (isobe-h) [inline lib/src/blocs/map_bloc.dart]: :-1: 0x20000000する意味がわからない

File: lib/src/blocs/map_bloc.dart [hunk 3]
```diff
@@ -110,27 +107,64 @@ class MapBloc extends Bloc{
...
+    _polygons.clear();
+    _draftPolygons.clear();
+    _draftMarkers.clear();
+    setLayers.add(layers);
+  }
 
-    Area area = Area();
-    area.areaPointsStr = Area.pointsToString(polygon.points);
-    AreaRepository().addPoints("area0", area);  //tableName,Area
+  Future<void> removeAreaFromAreaName(String areaName) async{
```
Comment (isobe-h) [inline lib/src/blocs/map_bloc.dart]: fromだと別の意味になる
removeAreaByAreaName

File: lib/src/blocs/read_area_bloc.dart [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    });
+  }
+
+  ReadAreaBloc(){
+    updateAreaList();
+  }
+
+  @override
+  void dispose() {
+    // TODO: implement dispose
```
Comment (isobe-h) [inline lib/src/blocs/read_area_bloc.dart]: :-1:

File: test/area_test.dart [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import 'package:flutter_map_app/src/models/area.dart';
```
Comment (isobe-h) [inline test/area_test.dart:1]: 引数がnullや空のリストの場合などの異常系のテストを加える

File: lib/src/blocs/map_bloc.dart (hunk 1 above)
Comment (riku04) [reply lib/src/blocs/map_bloc.dart]: 削除

File: lib/src/blocs/map_bloc.dart (hunk 3 above)
Comment (riku04) [reply lib/src/blocs/map_bloc.dart]: usingに変更

File: lib/src/blocs/read_area_bloc.dart (hunk 4 above)
Comment (riku04) [reply lib/src/blocs/read_area_bloc.dart]: 削除

File: lib/src/blocs/map_bloc.dart (hunk 2 above)
Comment (riku04) [reply lib/src/blocs/map_bloc.dart]: Constantsにアルファマスク値として定数置いた

File: lib/src/blocs/map_bloc.dart (hunk 3 above)
Comment (riku04) [reply lib/src/blocs/map_bloc.dart]: 修正

**Comment by isobe-h** [comment]
:+1:

**Pull request merged**